documentation/Tutorials/WPF/ConditionalExecution.md

```diff
@@ -0,0 +1,186 @@
+
+### Completing the conditions
+
+Using the same principle, execution conditions are added for the Delete and Save button. The ``CanSave`` property checks if you actually created a ``CategoryName`` to avoid the possibility to save empty Categories. Note you need to name sure ``CategorName`` is not null as well.
+
+The code for the complete ``CategoryViewModel`` look like this, with the last updates added:
+
+```csharp
+using Caliburn.Micro.Tutorial.Wpf.Models;
+
+namespace Caliburn.Micro.Tutorial.Wpf.ViewModels
+  {
+  public class CategoryViewModel : Screen
+    {
+    private BindableCollection<CategoryModel> _categoryList = new BindableCollection<CategoryModel>();
+    private CategoryModel _selectedCategoryModel;
+    private string _categoryName;
+    private string _categoryDescription;
+
+    public BindableCollection<CategoryModel> CategoryList
+      {
+      get
+        {
+        return _categoryList;
+        }
+      set
+        {
+        _categoryList = value;
+        }
+      }
+
+    public CategoryModel SelectedCategory
+      {
+      get
+        {
+        return _selectedCategoryModel;
+        }
+
+      set
+        {
+        _selectedCategoryModel = value;
+        NotifyOfPropertyChange(() => SelectedCategory);
+        NotifyOfPropertyChange(() => CanEdit);
+        NotifyOfPropertyChange(() => CanDelete);
+        }
+      }
+
+    public string CategoryName
+      {
+      get => _categoryName; set
+        {
+        _categoryName = value;
+        NotifyOfPropertyChange(() => CategoryName);
+        NotifyOfPropertyChange(() => CanSave);
+        }
+      }
+
+    public string CategoryDescription
+      {
+      get => _categoryDescription; set
+        {
+        _categoryDescription = value;
+        NotifyOfPropertyChange(() => CategoryDescription);
+        }
+      }
+
+    protected override void OnViewLoaded(object view)
+      {
+      base.OnViewLoaded(view);
+      CategoryList.Add(new CategoryModel { CategoryName = "Meals", CategoryDescription = "Lunched and diners" });
+      CategoryList.Add(new CategoryModel { CategoryName = "Representation", CategoryDescription = "Gifts for our customers" });
+      }
+
+
+    public bool CanEdit
+      {
+      get
+        {
+        return SelectedCategory != null;
+        }
+      }
+
+    public void Edit()
+      {
+      CategoryName = SelectedCategory.CategoryName;
+      CategoryDescription = SelectedCategory.CategoryDescription;
+      }
+
+    public bool CanDelete
+      {
+      get
+        {
+        return SelectedCategory != null;
+        }
+      }
+
+    public void Delete()
+      {
+      CategoryList.Remove(SelectedCategory);
+      Clear();
+      }
+
+    public bool CanSave
+      {
+      get
+        {
+        return CategoryName?.Length > 2;
+        }
+      }
+
+    public void Save()
+      {
+      CategoryModel newCategory = new CategoryModel();
+      newCategory.CategoryName = CategoryName;
```

Comment:
if make CategoryModel inherit from PropertyChangedBase you would not need to remove old category and add a new one in.

Reply:
That makes sense. I will not do this here, but there will be second user control in the tutorial. This allows readers to compare both solutions. I think you proposed solution has the disadvantage that the the UI aspects would be injected into the data logic.  In this case I probably would use two separate datamodels, one specifically for the UI layer and one for the data access layer. In the present example I am smuggling a bit with this additional complexity. I know tools like AutoFac would help to make life a bit easer here, but I would like to keep that for later, if I want to cover that.